add nginxModules.brotli

knupfer opened the pull request:
This adds the nginx module for static and on the fly brotli compression. It needs libbrotli which needs a newer version of brotli. Because of that, brotliUnstable is included.
On my side of the internet, all packages (including nginx with the module) compile fine.

abbradar commented:
Please split the commit into separate ones for each package, with commit messages according to our guidelines. For example, `brotliUnstable: init at 0.3.0`, `nginxModules.brotli: init at 788615e` (revision is fine if we don't have a specific version) and so on.

copumpkin commented:
It seems like `brotliUnstable` could simply be `brotli.overrideDerivation { src = ... }`, given that (I think) it's identical outside of the source link.

abbradar commented:
Hm, I remember vaguely that we discourage use of `overrideDerivation` in nixpkgs, but I can't remember for sure.

copumpkin commented:
Perhaps, but I also hate copying and pasting code :smile:

It doesn't seem like the usual concerns about `overrideDerivation` blowup would really apply here.

vcunat commented:
It isn't even 24h since I pushed brotli :-)

What perplexes me about libbrotli is that the github repo and homepage seem not to exist (anymore).

vcunat commented:
Moreover, you only use `src` from `brotliUnstable` so I'm not sure if it's worth to keep it around.

abbradar commented:
@knupfer One thing that I've failed to mention -- if there is a `name` attribute where you can actually specify some version (e.g. `nginx-1.8.0`) we prefer to have dates of the source revision there. For example, if you use revision `123bcde` which was committed at 2015-12-24 for package `nginx`, the name would be `nginx-20151224`. This applies to `brotliUnstable` and `libbrotli`, but not to `nginxModules.brotli` since there is no place to specify a version there.

knupfer commented:
@vcunat the repo and homepage exist fine at https://github.com/bagder/libbrotli
Actually I was writing the nix brotli exactly the same time you pushed yours (I finished 30 min after your push). I would also say, that brotliUnstable is useless, should we just raise the version of brotli from v0.3 to head? Brotli isn't finalized, so there's actually no point in having a "stable" version.

knupfer commented:
Oh, in there is actually written badger and not bagder in the homepage link. (It's the author of curl)

knupfer committed:
brotliUnstable: init at bed93862

knupfer committed:
libbrotli: init at 53d53e8

knupfer committed:
nginxModules.brotli: init at 788615e

knupfer commented:
I've fixed the typo and changed the names. How should we proceed? Should I remove brotliUnstable and update brotli?
I've tested the module on my server and it works fine, albeit the algo is disappointingly slow for on the fly compression

abbradar commented:
LGTM now, let's resolve what should we do with `brotli` and merge. Thanks!

zimbatm commented:
@knupfer do you strictly need brotliUnstable with libbrotly ? Just checked but google hasn't made another official release since last month.

knupfer commented:
Yes, without it libbrotli doesn't compile. I would suggest that we use simply only the most recent brotli. The old version is as well a beta version, so it wouldn't be "unstabler".

zimbatm commented:
Alright. Nothing is depending on brotli so it's fine to go unstable I think. Ping me here or on IRC once you've done the change.

joachifm commented:
@knupfer ping

lethalman commented:
Travis passed, thanks.
